### PR TITLE
Add user confirmation to toggle favorite action

### PR DESCRIPTION
**Pull request by Nisha-QA**

### CODE REVIEW

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 1]
```diff
@@ -48,13 +52,22 @@ suite('nuxeo-favorites-toggle-button', () => {
...
+        window.confirm.restore();
+      }
+    });
+
     test('it should display the document as favorite', () => {
       expect(element.favorite).to.be.true;
     });
 
-    test('toggle should remove document from favorites', async () => {
+    test('toggle should remove document from favorites when clicked ok', async () => {
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:64]: We should also check what happens when the user clicks cancel.

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 2]
```diff
@@ -84,13 +97,20 @@ suite('nuxeo-favorites-toggle-button', () => {
...
+        window.confirm.restore();
+      }
+    });
 
     test('it should display the document as not favorite', () => {
       expect(element.favorite).to.be.false;
     });
 
-    test('toggle should add the document to favorites', async () => {
+    test('toggle should add the document to favorites when clicked on Ok', async () => {
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:116]: We should also check what happens when the user clicks cancel.

File: ui/actions/nuxeo-favorites-toggle-button.js [hunk 3]
```diff
@@ -147,25 +147,29 @@ import '../nuxeo-button-styles.js';
...
         this.$.opAdd.execute().then(() => {
-          this.dispatchEvent(
-            new CustomEvent('added-to-favorites', {
-              composed: true,
-              bubbles: true,
-              detail: { doc: this.document },
-            }),
-          );
-          this._setFavorite(true);
+          if (window.confirm('Are you sure you want to add this document to Favorites?') === true) {
```
Comment (Gabez0r) [inline ui/actions/nuxeo-favorites-toggle-button.js]: This label should be internationalized.

File: ui/actions/nuxeo-favorites-toggle-button.js [hunk 4]
```diff
@@ -147,25 +147,29 @@ import '../nuxeo-button-styles.js';
...
         this.$.opRemove.execute().then(() => {
-          this.dispatchEvent(
-            new CustomEvent('removed-from-favorites', {
-              composed: true,
-              bubbles: true,
-              detail: { doc: this.document },
-            }),
-          );
-          this._setFavorite(false);
+          if (window.confirm('Are you sure you want to remove this document from Favorites?') === true) {
```
Comment (Gabez0r) [inline ui/actions/nuxeo-favorites-toggle-button.js]: This label should be internationalized.

**Comment by richardsd** [comment]
Nice work Nisha 👍 . Please review Gabriel's commits

**Review by Gabez0r — COMMENTED** [review]
Well done! 💪

**Comment by Gabez0r** [comment]
However, we do have a bunch of commits are not meaningful to the history (e.g., fixing linting, removing spaces, etc), so we should squash them.

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 5]
```diff
@@ -48,16 +52,31 @@ suite('nuxeo-favorites-toggle-button', () => {
...
       tap(element);
       await waitChanged(element, 'favorite');
       expect(element.favorite).to.be.false;
     });
+
+    test('toggle should not remove document from favorites when clicked Cancel', async () => {
+      // Remove document from favorites by toggling
+
+      sinon.stub(window, 'confirm').returns(false);
+      expect(element.favorite).to.be.true;
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:78]: Didn't we forget to actually click the button before doing the assertion?

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 6]
```diff
@@ -84,16 +103,30 @@ suite('nuxeo-favorites-toggle-button', () => {
...
       tap(element);
       await waitChanged(element, 'favorite');
       expect(element.favorite).to.be.true;
     });
+
+    test('toggle should not add the document to favorites when clicked on Cancel', async () => {
+      // Add the documents to favorites by toggling
+
+      sinon.stub(window, 'confirm').returns(false);
+      expect(element.favorite).to.be.false;
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:130]: Didn't we forget to actually click the button before doing the assertion?

**Review by Gabez0r — COMMENTED** [review]
Please see the latest comments in https://github.com/nuxeo/nuxeo-elements/pull/539

**Review by Gabez0r — COMMENTED** [review]
Well done! So far, so good.

File: ui/i18n/messages.json [hunk 7]
```diff
@@ -242,6 +244,7 @@
   "operationButton.bulk.poll.error": "Error while executing {0}",
   "operationButton.bulk.poll.running": "{0} is running ({1}/{2})",
   "operationButton.bulk.poll.scheduled": "{0} is scheduled. You will be notified when it starts.",
+  "parentInspectorButton.tooltip": "Parent Inspector",
```
Comment (Gabez0r) [inline ui/i18n/messages.json]: I see the label in this repository, but I don't see where it is used. If the element is on web ui, and the label is only used there, then it should be on the nuxeo web ui repository as well.

**Review by Gabez0r — COMMENTED** [review]
Looks good.

**Review by richardsd — COMMENTED** [review]
Looks good to me 👍 Please review the check https://github.com/nuxeo/nuxeo-elements/runs/7106071766?check_suite_focus=true because there are some linting issues.

**Comment by Nisha-QA** [comment]
Closing as the PR was part of exercise

**Pull request closed without merging**